server/filestore.go

```diff
@@ -8534,7 +8555,15 @@ func (fs *fileStore) flushStreamStateLoop(qch, done chan struct{}) {
 	for {
 		select {
 		case <-t.C:
-			fs.writeFullState()
+			err := fs.writeFullState()
+			if isPermissionError(err) {
+				fs.warn("File system permission denied when flushing stream state, disabling jetstream %v", err)
+				// messages in block cache could be lost in the worst case.
+				// In the clustered mode it is very highly unlikely as a result of replication.
+				fs.srv.DisableJetStream()
```

Comment:
I don't _think_ `fs.srv` can be `nil` here but let's add a `fs.srv != nil` guard just in case.